packages/core/src/auth/tokens/token-scope.ts

```diff
@@ -0,0 +1,35 @@
+export enum CacheRole {
+  None = 'none',
+  ReadWrite = 'readwrite',
+}
+
+export interface CacheRestriction {
+  cacheRole: CacheRole;
+}
+
+export enum TopicRole {
+  None = 'none',
+  ReadWrite = 'readwrite',
+}
+
+export interface TopicRestriction {
+  topicRole: TopicRole;
+}
+
+export type Restriction = CacheRestriction | TopicRestriction;
+
+export interface Restrictions {
+  restrictions: Array<Restriction>;
+}
+
+export class AllDataReadWrite implements Restrictions {
```

Comment:
is it better to `export const ALL_DATA_READ_WRITE: Restrictions = { restrictions: [ ... ] }`?

Reply:
> So e.g. I could create a class called `SingleCacheReadOnly` and then have it take a constructor arg `cacheName`.

Just want to also call out that we will probably want to support a case where we want something like `SingleCacheReadOnly` for cache foo but `SingleCacheReadWrite` for cache bar.

Reply:
seems to me like a builder that returns a `Restrictions` after all the optional friendly named configurables were provided would be the future I'd steer toward. Constants don't seem too terribly incongruous with builders in my experience, so that's where my head has been